explainaboard/utils/eval_utils/qa_table_text_hybrid.py

```diff
@@ -0,0 +1,328 @@
+The pre-processing is so complicated that I put most of the original processing code
+here. The official evaluation script can be found here:
+https://github.com/NExTplusplus/L2I/blob/main/evaluate.py
+"""
+
+from __future__ import annotations
+
+import re
+import string
+from typing import Union
+
+import numpy as np
+from scipy.optimize import linear_sum_assignment
+
+
+def negative_num_handle(x: str):
+    """
+    :param x:  transform (134) -> -134
+    :return:
+    """
+    all = re.findall('(\([\d.\s]+\))', x.strip())  # noqa: W605
+    if len(all) > 0:
+        return -1
+    return 1
+
+
+def percent_num_handle(x: str):
+    """
+    :param x:  transform 12% -> 12/100
+    :return:
+    """
+    all = re.findall('([\d.\s]+%)', x.strip())  # noqa: W605
+    if len(all) > 0:
+        return 0.01
+    return 1
+
+
+def word_scale_handle(x: str):
+    """
+    :param x: 1 million = 1,000,000
+    :return:
+    """
+    iter = re.finditer('([\d.]+\s?[a-zA-Z]+)', x)  # noqa: W605
+    for one in iter:
+        text = one.group(0).lower()
+        scale_val = scale_to_num(text)
+        return scale_val
+    return 1
+
+
+def scale_to_num(scale: str):
+    scale = scale.lower()
+    num = 1.0
+    if 'hundred' in scale:  # hundred
+        num = 100.0
+    elif 'thousand' in scale:  # thousand
+        num = 1000.0
+    elif 'million' in scale:  # million
+        num = 1000000.0
+    elif 'billion' in scale:  # billion
+        num = 1000000000.0
+    elif 'percent' in scale:  # percent
+        num = 0.01
+    return num
+
+
+def extract_one_num_from_str(s):
+    s = _clean_num(s)
+    r_num = r"([+-]?\d+(\.\d+)?)|([+-]?\.\d+)"
+    groups = re.findall(r_num, s)
+    if len(groups) == 0:
+        return None
+    num = groups[0][0]
+    if num == '':
+        return None
+    if '.' in num:
+        return float(num)
+    return int(num)
+
+
+EXCLUDE_IN_NUM = "'\"\\$€£¥%(),[]"
+
+
+def _clean_num(text: str):
+    return "".join([ch for ch in str(text) if ch not in EXCLUDE_IN_NUM])
+
+
+def is_number(text: str) -> bool:
+    try:
+        words = " ".join([_clean_num(w) for w in text.split()]).split()
+        if len(words) == 0:
+            """1023 or 1 million"""
+            return False
+        num = float(words[0])
+        if np.isnan(num):
+            return False
+        if len(words) >= 2:
+            if scale_to_num(words[1]) == 1:
+                return False
+        return True
+    except ValueError:
+        return False
+
+
+def to_number(text: str) -> float | None:
+    num = extract_one_num_from_str(text)
+    scale_val = word_scale_handle(text)
+    negative_flag = negative_num_handle(text)
+    percent_flag = percent_num_handle(text)
+    if num is not None:
+        return round(num * scale_val * negative_flag * percent_flag, 4)
+    return None
+
+
+def remove_articles(text: str) -> str:
+    regex = re.compile(r'\b(a|an|the)\b', re.UNICODE)
+    return re.sub(regex, ' ', text)
```

Comment:
It looks there is no reason to precomplie the pattern in the same function.
```suggestion
    return re.sub(r'\b(a|an|the)\b', ' ', text)
```